File: react-native/test/src/pages/home/ui/components/detailUser.js
import React, { memo } from 'react'
import { View } from 'react-native'
import { List } from 'react-native-paper'

const { Item, Icon } = List

const DetailUser = ({ dataUser }) => {

  return (
    <View>
      <Item
        title={dataUser?.name}
        description={dataUser?.email}
        left={({ color }) => <Icon color={color} icon="account" />}
      />
      <Item
        title='City'
        description={dataUser?.address?.city || ''}
        left={({ color }) => <Icon color={color} icon="city" />}
      />
      <Item
        title='Company'
        description={dataUser?.company?.name || ''}
        left={({ color }) => <Icon color={color} icon="home-city" />}
      />
      <Item
        title='Cellphone'
        description={dataUser?.phone || ''}
        left={({ color }) => <Icon color={color} icon="phone" />}
      />
    </View>
  )
}

export default memo(DetailUser)

File: flutter/app_test/lib/pages/home/ui/components/detailUser.dart
import 'package:app_test/pages/home/ui/components/itemList.dart';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:app_test/pages/home/providers/home.dart';

class DetailUser extends StatelessWidget {
  const DetailUser({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final detailUser = context.watch<HomeProvider>().detailUser;

    return Padding(
      padding: EdgeInsets.only(
        top: 5,
        bottom: MediaQuery.of(context).padding.bottom,
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          ItemList(
            title: detailUser?.name,
            subTitle: detailUser?.email,
            iconLeading: Icons.account_circle_rounded,
          ),
          ItemList(
            title: 'City',
            subTitle: detailUser?.address?.city,
            iconLeading: Icons.location_city,
          ),
          ItemList(
            title: 'Company',
            subTitle: detailUser?.company?.name,
            iconLeading: Icons.home_work_sharp,
          ),
          ItemList(
            title: 'Phone',
            subTitle: detailUser?.phone,
            iconLeading: Icons.phone,
          ),
        ],
      ),
    );
  }
}
